platforms/ide/tooling-api/src/main/java/org/gradle/tooling/events/test/ResourceBasedJvmTestOperationDescriptor.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling.events.test;
+
+import org.gradle.api.Incubating;
+import org.gradle.tooling.events.test.testsource.FilesystemSource;
+import org.jspecify.annotations.NullMarked;
+
+/**
+ * Describes a resource-based test that runs on the JVM and for which an event has occurred.
+ *
+ * @since 9.4.0
+ */
+@NullMarked
+@Incubating
+public interface ResourceBasedJvmTestOperationDescriptor extends TestOperationDescriptor {
```

Comment:
Do we actually care about this distinction (resource v class)?  How can we reliably determine if something is a class vs a resource?

e.g., `TestSource` is extensible, I think, so any type we do not recognize will fall into our Unknown test source.  So we could have "resource tests" that have unknown test sources and then we'd report those as JvmTestOperationDescriptors.

Is the `JvmTestKind` important here?